site/_posts/2023-01-12-what-happens-after-authentication-event.md

```diff
@@ -0,0 +1,136 @@
+
+If the account is still good and the user is still logged in, the auth server can issue a new access token, transparently extending your application's session.
+
+### Benefits Of Client Stored Tokens
+
+If you choose this path, you gain horizontal scalability. As long as your APIs are on the same base domain, they are presented with any request your application makes. 
+
+As mentioned above, this approach is a perfect fit for a JavaScript, single page application pulling data from multiple APIs hosted by the same organization.
+
+Using cookies means you are safe from XSS attacks, a common mechanism for attackers to gain access to tokens and to make requests masquerading as another user. `HTTPOnly`, secure cookies are not available to any JavaScript running on the page, so can't be accessed by malicious scripts.
+
+If the APIs are on different domains, either use a proxy which can ingest the token, validate it and pass on requests to other domains, or use the session based approach, discussed later. Below is a diagram of using the proxy approach, where an API from `todos.com` is called through a proxy living at `proxy.example.com`.
+
+{% plantuml source: _diagrams/blogs/after-authorization-code-grant/client-side-storage-with-proxy.plantuml, alt: "Using a proxy to access APIs on different domains." %}
+
+### Alternatives To Client Stored Tokens In the Browser
+
+Why use a browser cookie and not another storage mechanism such as memory or localstorage? Why not bind the cookie to the browser? All options have tradeoffs, and using cookies works for most of our customers.
+
+Localstorage is a difficult option because, unless you have fingerprint your token, as [recommended by OWASP](https://cheatsheetseries.owasp.org/cheatsheets/JSON_Web_Token_for_Java_Cheat_Sheet.html#token-sidejacking), you'll be prone to those XSS attacks, since all JavaScript running on the page has access to the token. If you fingerprint your token, you'll be limited to sending it to APIs running on the domain the cookie is sent to, anyway.
+
+If you use an in-memory storage solution, when the browser is refreshed, the token goes poof. The user has to log in again, which is not a great experience.
+
+Client binding measures remove the danger of XSS because the token can't be used without a private key that only the proper client possesses. However, these approaches require additional setup on the client-side and are relatively new; for example, as of this writing, DPoP is not yet an IETF standard. 
+
+## Using Sessions
+
+If client-side storage doesn't meet your needs, another option is to store the access token and refresh token in the server side session. The application can then use normal web sessions.
+
+{% plantuml source: _diagrams/blogs/after-authorization-code-grant/session-storage.plantuml, alt: "Storing the tokens server-side in a session." %}
```

Comment:
The other diagrams reference the first diagram.  It'd be helpful for this diagram to also reference the first diagram (and be clear that it's referenced) for consistency.